lib/lightning/maintenance/partition_table_service.ex

```diff
@@ -0,0 +1,182 @@
+defmodule Lightning.PartitionTableService do
+  @moduledoc """
+  Service to keep the partition tables up to date.
+  """
+
+  use Oban.Worker,
+    queue: :background,
+    max_attempts: 1
+
+  import Ecto.Query
+
+  alias Lightning.Repo
+
+  require Logger
+
+  @impl Oban.Worker
+  def perform(%Oban.Job{args: %{"add_headroom" => %{"weeks" => weeks}}})
+      when is_integer(weeks) do
+    add_headroom(:all, weeks)
+  end
+
+  @impl Oban.Worker
+  def perform(%Oban.Job{args: %{"drop_older_than" => %{"weeks" => weeks}}})
+      when is_integer(weeks) do
+    upper_bound = Timex.shift(DateTime.utc_now(), weeks: weeks)
```

Comment:
Could you try using `DateTime.shift` instead of Timex - nothing against what you've done here - Timex was the gold standard for all things Date but the built in DateTime has come a long way since then. And we have had issues with Timex in compiled environments (like without having the entire app running, like in migrations on production); if you can achieve a passing test suite by just replacing the use of Timex then we should be good.

Reply:
@stuartc I have expunged all the uses of `Timex` from prod code, with one [exception](https://github.com/OpenFn/Lightning/pull/1392/files#diff-8fde6fc40ea0bdb0e15ecfaa85a407a45cbc6db9f74137f45d64bc7b8bfc1609R7) - I can't find an equivalent to `iso_week` on either `Date` or `DateTime` - if you think this is going to be problematic, I can try and handroll an equivalent?